### DISCUSSION TITLE
Response curves - help required plotting from 'first principles'

### DISCUSSION THREAD

**Discussion by gedkitchener-CT** [facebookexperimental/Robyn#328] [Q&A] [answered]
**Approach A**
From the `pareto_media_transform_matrix.csv` file, I am plotting `decompMedia` (y) vs. (x) `adstockedMedia` for a paid-media variable (for a single `solID`).

**Approach B**
Using the Hill Equation. With `y = β (x^α / (x^α + γ^α))`, I get the `β`, `α`, and `γ` values for the relevant paid-media variable and model, and plug them into the equation.

For example, with `β = 50,000`, `α = 3.7`, and `γ = 0.6`, I find that the fraction very quickly tends to 1.0 because `x^α >> γ^α` (even for relatively small values of `x`).

**Issue**
The two approaches (A and B) don't agree - have I misunderstood how the Hill equation is used perhaps? Any help would be great.

**Comment by gedkitchener-CT** [comment]
For example, is `gamma` scaled in some way? (e.g., by max of `x`)

**Reply by igorskokan** [reply]
Hey! γ is the inflection parameter in Hill function. 
What units is x in? What are the typical values of x - is it very small, e.g. x ∈ <0,1> for example?

**Reply by gedkitchener-CT** [reply]
`x` is impressions or spend, so runs in the thousands, while `γ` is bounded by `(0.3, 1.0)`, as per the suggestion in the [docs](https://facebookexperimental.github.io/Robyn/docs/analysts-guide-to-MMM/#:~:text=We%20recommend%20a%20bound%20of%20c(0.3%2C%201)).

This is why I'm confused - there's something I'm missing in how the Hill function is actually applied, i.e., `γ` values between `(0.3, 1.0)` cannot be appropriate for typical impression volumes or spend levels (what marketing team would aim for on the order of 1 impression per day?!?)

Hence my question about whether `γ` is perhaps scaled behind-the-scenes, to make it roughly the same magnitude as the metric of interest (e.g., spend level)?

**Reply by gedkitchener-CT** [reply]
As it turns out, my suspicions were confirmed.

But rather than `γ` being scaled, it is `x` that is scaled.

**Reply by igorskokan** [reply]
Thanks @gedkitchener-CT  apologies if I misunderstood your original question. As per your own investigation the Hill saturation transformation normalises all independent vars to 0-1

**Comment by gedkitchener-CT** [comment, marked as answer]
Oh, I think I've found the answer in another [discussion](https://github.com/facebookexperimental/Robyn/discussions/291#:~:text=as%20%22weekly%20contribution%22.-,Finally%2C%20all%20independent%20variables%20are%20transformed%20before%20the%20fitting%20as%20you%20correctly%20acknowledged.%20Because%20the%20hill%20function%20we%20use%20for%20saturation%20transformation%20normalises%20all%20independent%20to%200%2D1%2C%20that%27s%20why%20beta%20coefficients%20gets%20bigger.,-Marked%20as%20answer)

> all independent variables are transformed before the fitting ... the hill function we use for saturation transformation normalises all independent to 0-1

**EDIT**

Actually, it seems that it's indeed `γ` that is transformed (see [transformation.R](https://github.com/facebookexperimental/Robyn/blob/6ab4323b5e5305df6f6bcc600fba060b38d18ee8/R/R/transformation.R#:~:text=gammaTrans%20%3C%2D%20round(quantile(seq(range(x)%5B1%5D%2C%20range(x)%5B2%5D%2C%20length.out%20%3D%20100)%2C%20gamma)%2C%204))). That is to say, it's not `x` that is transformed.

`gammaTrans <- round(quantile(seq(range(x)[1], range(x)[2], length.out = 100), gamma), 4)`
